src/renderer/coremods/utilityClasses/index.ts

```diff
@@ -0,0 +1,138 @@
+
+const inject = new Injector();
+const html = document.documentElement;
+
+interface TabBarItemProps {
+  id: string;
+  "aria-controls"?: string;
+}
+
+interface TabBarItemType extends React.Component<TabBarItemProps> {
+  render(): React.ReactElement<TabBarItemProps>;
+}
+
+// Re-adds the tab bar item's ID as an always-present attribute
+function tabBarItemId(): void {
+  const TabBarModule = getByProps<{ TabBar: { Item: { prototype: TabBarItemType } } }>("TabBar");
+  if (!TabBarModule) {
+    throw new Error("Failed to find TabBar module!");
+  }
+
+  inject.after(
+    TabBarModule.TabBar.Item.prototype,
+    "render",
+    function (this: TabBarItemType, _, res) {
+      if (typeof this.props.id === "string") {
+        res.props["aria-controls"] = `${this.props.id.replace(/\s+/g, "-").toLowerCase()}-tab`;
+      }
+      return res;
+    },
+  );
+}
+
+interface ClientThemesBackgroundStore extends Store {
+  gradientPreset: { id: number } | undefined; // undefined when no nitro theme is selected
+}
+type ThemeIDMap = Record<string, number> & Record<number, string>;
+
+function onNitroThemeChange(store: ClientThemesBackgroundStore, ThemeIDMap: ThemeIDMap): void {
+  if (!store.gradientPreset) {
+    html.removeAttribute("data-nitro-theme");
+  } else {
+    const theme = ThemeIDMap[store.gradientPreset.id];
+    html.setAttribute("data-nitro-theme", theme);
+  }
+}
+
+// Adds the currently active nitro theme as a class on the html element
+function nitroThemeClass(): void {
+  const ClientThemesBackgroundStore = getByStoreName<ClientThemesBackgroundStore>(
+    "ClientThemesBackgroundStore",
+  );
+  if (!ClientThemesBackgroundStore) {
+    throw new Error("Failed to find ClientThemesBackgroundStore!");
+  }
+  const ThemeIDMap = getByProps<ThemeIDMap>("MINT_APPLE");
+  if (!ThemeIDMap) {
+    throw new Error("Failed to find ThemeIDs module!");
+  }
+
+  // update theme attribute when theme changes
+  ClientThemesBackgroundStore.addChangeListener(() => {
+    onNitroThemeChange(ClientThemesBackgroundStore, ThemeIDMap);
+  });
+  onNitroThemeChange(ClientThemesBackgroundStore, ThemeIDMap);
+}
+
+interface MessageComponent {
+  props: { id: string | undefined }; // may not be there for non-message components
+  type: {
+    type: (msg: { message: Message }) => React.ReactElement;
+  };
+}
+
+async function messageDataAttributes(): Promise<void> {
+  const MessagesComponent = await waitForModule<{
+    type: () => React.ReactElement;
+  }>(filters.bySource(".content.id)"));
+
+  // the Message component isn't exported, so it must be extracted like this
+  const uninjectMessagesComponent = inject.after(MessagesComponent, "type", (_, res) => {
+    uninjectMessagesComponent();
+
+    const MessageListComponent = res.props.children.type;
+    const uninjectMessageList = inject.after(MessageListComponent, "type", (_, res) => {
+      const messagesArray = res.props.children.props.children[1].props.children[1].props
```

Comment:
This line errored when I tried to view a channel. Please add handling to this so it won't crash.

Reply:
this got removed because the webpack search was made a lot easier, but i made sure the new one has adequate error checking.